streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/StatefulProcessorNode.java

```diff
@@ -92,24 +94,24 @@ private StatefulProcessorNodeBuilder() {
             return this;
         }
 
-        public StatefulProcessorNodeBuilder<K, V> withStoreBuilder(final StoreBuilder<KeyValueStore<K, V>> storeBuilder) {
+        public StatefulProcessorNodeBuilder<K, V> withStoreBuilder(final StoreBuilder<? extends StateStore> storeBuilder) {
             this.storeBuilder = storeBuilder;
             return this;
         }
 
-        public StatefulProcessorNodeBuilder<K, V> withMaybeRepartitionedSourceName(String maybeRepartitionedSourceName) {
+        public StatefulProcessorNodeBuilder<K, V> withMaybeRepartitionedSourceName(final String maybeRepartitionedSourceName) {
             this.maybeRepartitionedSourceName = maybeRepartitionedSourceName;
             return this;
         }
 
         public StatefulProcessorNode<K, V> build() {
             return new StatefulProcessorNode<>(nodeName,
-                                               processorSupplier,
-                                               storeNames,
-                                               maybeRepartitionedSourceName,
-                                               storeBuilder,
-                                               repartitionRequired);
-
+                processorSupplier,
```

Comment:
This seems inconsistent as your principle across multiple PRs is to have new line for first parameter. I'm not completely in favor of doing so (see my comment on your other PR) but just pointing it out. Please lmk if we really need this to avoid compilation warnings.